data/mods/Aftershock/maps/mapgen/standard_colonial/standard_habitat_mech.json

```diff
@@ -0,0 +1,207 @@
+[
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "afs_mech_standard_habitat" ],
+    "weight": 500,
+    "object": {
+      "fill_ter": "t_metal_floor",
+      "rows": [
+        "1                       ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        "
+      ],
+      "terrain": { "/": "t_wall_prefab_metal" },
+      "place_nested": [
```

Comment:
These hundred odd lines of manual chunk placement depending on rotation are excedingly ugly, and dont provide excellent results when the orientations of neighbors are mixed (ex: one facing _north and the other one _south). 

 But the way these chunks are rotated is relatively custom and I failed to conceive the way to do it with c++, help would be appreciated if someone is up to it